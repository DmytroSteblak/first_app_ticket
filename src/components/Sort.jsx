import React from 'react';
import { useDispatch } from 'react-redux';

import { ButtonComp } from './index';
import { fastTickets, sortTickets } from '../redux/ducks/TicketsReducer';
import { FilterRow } from '../styled/Filter-component';

const Sort = () => {
    const dispatch = useDispatch();
    const handleCheap = () => {
        dispatch(sortTickets());
    };
    const handleFast = () => {
        dispatch(fastTickets());
    };

    return (
        <FilterRow>
            <ButtonComp onClick={handleCheap}>
                самый дешевый
            </ButtonComp>
            <ButtonComp onClick={handleFast}>
                самый быстрый
            </ButtonComp>
        </FilterRow>
    );
};

export default Sort;