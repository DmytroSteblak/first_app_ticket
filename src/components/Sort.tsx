import React, { FC } from 'react';
import { useDispatch } from 'react-redux';

import { ButtonComp } from './index';
import { fastTickets, sortTickets } from '../redux/ducks/TicketsReducer';

import { FilterRow } from '../styled/Filter-component';

const Sort: FC = () => {
    const dispatch = useDispatch();
    const SortByPrice = () => {
        dispatch(sortTickets());
    };
    const SortBySpeed = () => {
        dispatch(fastTickets());
    };

    return (
        <FilterRow>
            <ButtonComp onClick={SortByPrice}>
                самый дешевый
            </ButtonComp>
            <ButtonComp onClick={SortBySpeed}>
                самый быстрый
            </ButtonComp>
        </FilterRow>
    );
};

export default Sort;