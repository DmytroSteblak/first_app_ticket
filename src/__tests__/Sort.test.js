import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';

import Sort from '../components/Sort';

const props = {
    onClick: jest.fn()
}

let store, component;
beforeEach(() => {
    component = mount(
        <Provider store={store}>
            <Sort {...props} />
        </Provider>
    );
});

describe('Sort component', () => {
    const initialState = {};
    const mockStore = configureStore();
    store = mockStore(initialState);

    it('should render Sort component', () => {
        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render two button',() => {
        const wrapper = component.find('ButtonComp');
        expect(wrapper.length).toBe(2);
    });

});
