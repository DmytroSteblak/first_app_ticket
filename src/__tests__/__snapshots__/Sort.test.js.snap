// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort component should render Sort component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Sort
    onClick={[MockFunction]}
  >
    <styled.div>
      <div
        className="sc-jSFjdj htJIKz"
      >
        <ButtonComp
          onClick={[Function]}
        >
          <styled.div>
            <div
              className="sc-eCApnc fqVHnC"
            >
              <Button
                active={false}
                disabled={false}
                onClick={[Function]}
                variant="primary"
              >
                <button
                  className="btn btn-primary"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  самый дешевый
                </button>
              </Button>
            </div>
          </styled.div>
        </ButtonComp>
        <ButtonComp
          onClick={[Function]}
        >
          <styled.div>
            <div
              className="sc-eCApnc fqVHnC"
            >
              <Button
                active={false}
                disabled={false}
                onClick={[Function]}
                variant="primary"
              >
                <button
                  className="btn btn-primary"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  самый быстрый
                </button>
              </Button>
            </div>
          </styled.div>
        </ButtonComp>
      </div>
    </styled.div>
  </Sort>
</Provider>
`;
